File: Frontend/src/app/service/backend/admin.service.ts
import { Injectable } from '@angular/core';
import { environment } from '../../../environments/environments';
import { HttpClient } from '@angular/common/http';
import { lastValueFrom } from 'rxjs';
import { Report, UserForAdmin, UUID } from '../../types';

@Injectable({
  providedIn: 'root'
})
export class AdminService {

  private baseUrl = `${environment.url}admin`;

  constructor(private http: HttpClient) { }

  getAllUsers(): Promise<UserForAdmin[]> {
    return lastValueFrom(this.http.get<UserForAdmin[]>(`${this.baseUrl}/getAllUsers`));
  }

  getUserReports(id: UUID): Promise<Report[]> {
    return lastValueFrom(this.http.get<Report[]>(`${this.baseUrl}/getUserReports/${id}`));
  }

  getUserById(id: UUID): Promise<UserForAdmin> {
     return lastValueFrom(this.http.get<UserForAdmin>(`${this.baseUrl}/getUserById/${id}`));
  }

  reportDisable(id: UUID, message: string): void {
    lastValueFrom(this.http.post(`${this.baseUrl}/reportDisable/${id}`, message));
  }
}

File: Backend/src/main/java/com/example/r100/hop2school/service/AdminService.java
package com.example.r100.hop2school.service;

import com.example.r100.hop2school.dto.Report;
import com.example.r100.hop2school.dto.User;
import com.example.r100.hop2school.dto.UserForAdmin;
import com.example.r100.hop2school.entity.UserEntity;

import java.util.List;
import java.util.UUID;

public interface AdminService {
    List<UserForAdmin> getAllUsers();

    List<Report> getReportsByUserId(UUID userId);

    void reportDisable(UUID userId, String message);

    UserForAdmin getUserById(UUID id);
}
